hivemq-edge/src/main/java/com/hivemq/protocols/AdapterConfigAndTagsAndFieldMappings.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019-present HiveMQ GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.protocols;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.hivemq.adapter.sdk.api.config.ProtocolAdapterConfig;
+import com.hivemq.adapter.sdk.api.config.legacy.ConfigTagsTuple;
+import com.hivemq.adapter.sdk.api.config.legacy.LegacyConfigConversion;
+import com.hivemq.adapter.sdk.api.factories.ProtocolAdapterFactory;
+import com.hivemq.adapter.sdk.api.tag.Tag;
+import com.hivemq.configuration.entity.adapter.FieldMappingsEntity;
+import com.hivemq.persistence.fieldmapping.FieldMappings;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.hivemq.protocols.ConfigPersistence.TAG_MAPPING_KEY;
+
+public class AdapterConfigAndTagsAndFieldMappings {
+
+    private static final Logger log = LoggerFactory.getLogger(AdapterConfigAndTagsAndFieldMappings.class);
+
+    private final @NotNull ProtocolAdapterConfig adapterConfig;
+    private final @NotNull List<? extends Tag> tags;
+    private final @NotNull List<FieldMappings> fieldMappings;
+
+    public AdapterConfigAndTagsAndFieldMappings(
+            final @NotNull ProtocolAdapterConfig adapterConfig,
+            final @NotNull List<? extends Tag> tags,
+            final @NotNull List<FieldMappings> fieldMappings) {
+        Objects.requireNonNull(adapterConfig);
+        Objects.requireNonNull(tags);
+        this.adapterConfig = adapterConfig;
+        this.tags = tags;
+        this.fieldMappings = fieldMappings;
```

Comment:
Please make a null check for this field.

Reply:
actually we do not test fields that are annotated with @NotNull as we are warned when handing over a nullable field. Unless we ignore the warning no null can ever be handed over.